packages/enzyme-matchers/src/utils/html.js

```diff
@@ -20,9 +20,13 @@ function mapWrappersHTML(wrapper: EnzymeObject): Array<string> {
       : inst._currentElement.props;
     consoleObject.error = error;
 
-    const transformedProps = Object.keys(props).map(
-      key => `${key}="${props[key]}"`,
-    );
+    const transformedProps = Object.keys(props).map(key => {
+      try {
+        return `${key}="${props[key].toString()}"`;
+      } catch (e) {
+        return `${key}="[object Object]"`;
+      }
```

Comment:
Hey, sorry I never got around to answering on Proxy detection (hectic times XD), but yeah you're right on that you can't trivially detect them. I've gone ahead and rebased the pull request